### PR TITLE
Implement callback for trunk [WIP]

### PR DESCRIPTION
**Pull request by abelsiqueira**
Initial implementation of a callback strategy.
`callback-example` plots the 2d trace of trunk as a simple example.

trunk-on-rosenbrock

### CODE REVIEW

File: src/trunk.jl [hunk 1]
```diff
@@ -32,8 +33,10 @@ function trunk(::Val{:Newton},
     error("trunk should only be called for unconstrained problems. Try tron instead")
   end
 
+  wks = Dict{Symbol,Any}(:x => x)
```
Comment (dpo) [inline src/trunk.jl:36]: That's a lot of storage. For this to be practical, I think we should let the user choose to store x.

File: src/trunk.jl [hunk 2]
```diff
@@ -43,31 +46,32 @@ function trunk(::Val{:Newton},
   # Armijo linesearch parameter.
   β = eps(T)^T(1/4)
 
-  iter = 0
-  f = obj(nlp, x)
-  ∇f = grad(nlp, x)
+  wks[:iter] = 0
+  wks[:f] = obj(nlp, x)
+  wks[:∇f] = ∇f = grad(nlp, x)
```
Comment (dpo) [inline src/trunk.jl:51]: lots of storage here too

File: src/trunk.jl (hunk 1 above)
Comment (abelsiqueira) [reply src/trunk.jl:36]: These should be only pointers, so they should be light.

Comment (dpo) [reply src/trunk.jl:36]: Oh I see. Then 👍

**Comment by dpo** [comment]
I like this. It has the benefits of iterators while keeping the algorithm readable.

**Comment by abelsiqueira** [comment]
I'm closing this in favor of #113.

**Pull request closed without merging**